DataStorage.py

```diff
@@ -0,0 +1,99 @@
+from typing import Dict, List
+import operator
+import copy
+import math
+
+def remove_from_list(container, value):
+    try:
+        container.remove(value)
+    except ValueError:
+        pass
+    return container
+
+def pop_from_container(container, value):
+    try:
+        container.pop(value)
+    except ValueError:
+        pass
+    return container
+
+def update_dict(dictionary, entries):
+    dictionary.update(entries)
+    return dictionary
+
+# functions callable on storable data on the server by clients
+modify_functions = {
+    "add": operator.add,  # add together two objects, using python's "+" operator (works on strings and lists as append)
+    "mul": operator.mul,
+    "mod": operator.mod,
+    "max": max,
+    "min": min,
+    "replace": lambda old, new: new,
+    "default": lambda old, new: old,
+    "pow": operator.pow,
+    "floor": lambda value, _: math.floor(value),
+    "ceil": lambda value, _: math.ceil(value),
+    # bitwise:
+    "xor": operator.xor,
+    "or": operator.or_,
+    "and": operator.and_,
+    "left_shift": operator.lshift,
+    "right_shift": operator.rshift,
+    # lists/dicts
+    "remove": remove_from_list,
+    "pop": pop_from_container,
+    "update": update_dict,
+}
+
+class InvalidArgumentsException(Exception):
+    pass
+
+class DataStorage:
+    stored_data: Dict[str, object]
+
+    def __init__(self, stored_data: Dict[str, object]):
+        self.stored_data = stored_data
+
+    @staticmethod
+    def validate_and_get_key(set_cmd: Dict[str, object]) -> str:
+        try:
+            key = set_cmd["key"]

```

Comment:
```suggestion
            key = set_cmd["key"]
            if not isinstance(key, str):
                raise InvalidArgumentsException(f"key has to be a string")
```
Otherwise your typing falls apart